### PR TITLE
Implement searchFunctions in the threadfront

### PR DESCRIPTION
**Pull request by jaril**
Fix #3151.

### CODE REVIEW

File: src/devtools/client/debugger/src/components/QuickOpenModal.js [hunk 1]
```diff
@@ -33,12 +31,14 @@ import {
...
   formatSources,
-  formatProjectSymbols,
 } from "../utils/quick-open";
 import Modal from "./shared/Modal";
 import SearchInput from "./shared/SearchInput";
 import ResultList from "./shared/ResultList";
 import { trackEvent } from "ui/utils/telemetry";
+import { ThreadFront } from "protocol/thread";
+import { getSourceIDsToSearch } from "devtools/client/debugger/src/components/FullTextSearch/search";
+import { getGlobalFunctions, isGlobalFunctionsLoading } from "../reducers/ast";
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/QuickOpenModal.js:39]: do we need this?

import { ThreadFront } from "protocol/thread";
import { getSourceIDsToSearch } from "devtools/client/debugger/src/components/FullTextSearch/search";

File: src/devtools/client/debugger/src/components/QuickOpenModal.js [hunk 2]
```diff
@@ -59,7 +59,10 @@ function filter(values, query) {
 export class QuickOpenModal extends Component {
   constructor(props) {
     super(props);
-    this.state = { results: null, selectedIndex: 0 };
+    this.state = {
+      results: null,
+      selectedIndex: 0,
+    };
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/QuickOpenModal.js]: doesnt need to be in the constructor 
```
state = {
      results: null,
      selectedIndex: 0,
    };
```

File: src/devtools/client/debugger/src/components/QuickOpenModal.js [hunk 3]
```diff
@@ -453,11 +447,12 @@ function mapStateToProps(state) {
       ? !!getSourceContent(state, selectedSource.id)
       : undefined,
     symbols: formatSymbols(getSymbols(state, selectedSource)),
-    projectSymbols: selectProjectSymbols(state),
-    projectSymbolsLoading: getProjectSymbolsLoading(state),
     symbolsLoading: isSymbolsLoading(state, selectedSource),
     query: getQuickOpenQuery(state),
     searchType: getQuickOpenType(state),
+    sourcesById: getSources(state).values,
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/QuickOpenModal.js:448]: do we need `sourcesById`

File: src/devtools/client/debugger/src/utils/quick-open.js [hunk 4]
```diff
@@ -66,39 +66,36 @@ export function formatSourcesForList(source, tabUrls) {
 
 export function formatSymbol(symbol) {
   return {
-    id: `${symbol.name}:${symbol.location.start.line}`,
+    id: `${symbol.name}:${symbol.location.line}`,
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/utils/quick-open.js:69]: do we need column too?

**Review by jasonLaster — COMMENTED** [review]
couple nits

File: src/devtools/client/debugger/src/components/QuickOpenModal.js (hunk 1 above)
Comment (jaril) [reply src/devtools/client/debugger/src/components/QuickOpenModal.js:39]: Just deleted it 👍

File: src/devtools/client/debugger/src/utils/quick-open.js (hunk 4 above)
Comment (jaril) [reply src/devtools/client/debugger/src/utils/quick-open.js:69]: added

File: src/devtools/client/debugger/src/components/QuickOpenModal.js [hunk 5]
```diff
@@ -268,17 +262,24 @@ export class QuickOpenModal extends Component {
...
     }
 
     this.updateResults(e.target.value);
   };
 
+  getResults = () => {
+    const { results } = this.state;
+
+    return results;
+  };
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/QuickOpenModal.js:276]: lets remove getResults

File: src/devtools/client/debugger/src/components/QuickOpenModal.js [hunk 6]
```diff
@@ -18,13 +17,12 @@ import {
...
   getSelectedSource,
   getSourceContent,
-  getProjectSymbols,
   getSymbols,
   getTabs,
   getDisplayedSources,
   isSymbolsLoading,
-  getProjectSymbolsLoading,
   getContext,
+  getSources,
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/QuickOpenModal.js:25]: is this needed?

File: src/devtools/client/debugger/src/components/FullTextSearch/search.js [hunk 7]
```diff
@@ -57,7 +57,7 @@ export async function search(query, sourcesById, updateResults) {
   updateResults(() => ({ status: "DONE" }));
 }
 
-function getSourceIDsToSearch(sourcesById) {
+export function getSourceIDsToSearch(sourcesById) {
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/FullTextSearch/search.js:60]: can we move this to a util file

**Pull request merged**